qiskit/wrapper/credentials/_configrc.py

```diff
@@ -0,0 +1,138 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+Utilities for reading and writing credentials from and to configuration files.
+"""
+
+import os
+from ast import literal_eval
+from configparser import ConfigParser, ParsingError
+
+from qiskit import QISKitError
+from qiskit.backends.ibmq import IBMQProvider
+
+
+DEFAULT_QISKITRC_FILE = os.path.join(os.path.expanduser("~"),
+                                     '.qiskit', 'qiskitrc')
+
+
+def read_credentials_from_qiskitrc(filename=None):
+    """
+    Read a configuration file and return a dict with its sections.
+
+    Args:
+        filename (str): full path to the qiskitrc file. If `None`, the default
+            location is used (`HOME/.qiskit/qiskitrc`).
+
+    Returns:
+        dict: dictionary with the contents of the configuration file, with
+            the form::
+
+            {'provider_class_name': {'token': 'TOKEN', 'url': 'URL', ... }}
+
+    Raises:
+        QISKitError: if the file was not parseable. Please note that this
+            exception is not raised if the file does not exist (instead, an
+            empty dict is returned).
+    """
+    filename = filename or DEFAULT_QISKITRC_FILE
+    config_parser = ConfigParser()
+    try:
+        config_parser.read(filename)
+    except ParsingError as ex:
+        raise QISKitError(str(ex))
+
+    # Build the credentials dictionary.
+    credentials_dict = {}
+    for name in config_parser.sections():
+        single_credentials = dict(config_parser.items(name))
+        # TODO: 'proxies' is the only value that is a dict. Consider moving to
+        # json configuration or splitting into single keys manually.
+        if 'proxies' in single_credentials.keys():
+            single_credentials['proxies'] = literal_eval(
+                single_credentials['proxies'])
+        credentials_dict[name] = single_credentials
+
+    return credentials_dict
+
+
+def write_qiskit_rc(credentials, filename=None):
+    """
+    Write credentials to the configuration file.
+
+    Args:
+        credentials (dict): dictionary with the credentials, with the form::
+            {'provider_class_name': {'token': 'TOKEN', 'url': 'URL', ... }}
+        filename (str): full path to the qiskitrc file. If `None`, the default
+            location is used (`HOME/.qiskit/qiskitrc`).
+    """
+    filename = filename or DEFAULT_QISKITRC_FILE
+    # Create the directories and the file if not found.
+    os.makedirs(os.path.dirname(filename), exist_ok=True)
+
+    # Write the configuration file.
+    with open(filename, 'w') as config_file:
+        config_parser = ConfigParser()
+        config_parser.read_dict(credentials)
+        config_parser.write(config_file)
```

Comment:
Is the special case for `proxies` automatically handled?

Reply:
For writing, yes, see `test_store_credentials()` - and for reading see the `if` in `_configrc.py`. It would be nicer to generalize it, but seems something that can be tackled in future iterations.

Reply:
On a second  look, `verify` also needed converting to `bool`. I tweaked the function to mark the need in future updates.